hirosystems/stacks-blockchain-api/values.yaml

```
@@ -1605,5 +1603,17 @@ postgresql:
     #       REVOKE ALL ON SCHEMA public FROM public;
     #       REVOKE ALL ON SCHEMA stacks_blockchain_api FROM public;
     #       GRANT ALL ON SCHEMA stacks_blockchain_api TO postgres;
+    # initContainers:
+    #   - name: pg-init-helper
+    #     image: google/cloud-sdk:alpine
+    #     command: [ "/bin/bash", "-c", "--" ]
+    #     args: [ "curl https://archive.stg.hiro.so/mainnet/stacks-blockchain-api-pg/stacks-blockchain-api-pg-15-latest.dump -o /bitnami/postgresql/stacks-blockchain-api-postgres.dump" ]
```

Maybe this should first check if there are any files in this directory first, and only download the dump when it's empty? That way the user wouldn't necessarily need to remove the initContainer on subsequent reboots if they didn't want to.

Also let's use the prd archive here.
```suggestion
    #     args: [ "curl https://archive.hiro.so/mainnet/stacks-blockchain-api-pg/stacks-blockchain-api-pg-15-latest.dump -o /bitnami/postgresql/stacks-blockchain-api-postgres.dump" ]
```

---

> Maybe this should first check if there are any files in this directory first, and only download the dump when it's empty? That way the user wouldn't necessarily need to remove the initContainer on subsequent reboots if they didn't want to.
> 
> Also let's use the prd archive here.

good idea, to further improve on this, I'm thinking we use the same logic for `0_restore_archive.sh`, what do you think?

run pg_restore only if a `*.dump` file is found; I feel `1_configure_schema.sql` is idempotent as it's setting database permissions, so it shouldn't have negative affects.

---

Bitnami actually has that logic already baked in to their image. It basically checks to see if the database has any data. If it doesn't, it runs the init scripts, otherwise it doesn't run them again.